crates/engine/primitives/src/lib.rs

```diff
@@ -106,7 +130,7 @@ pub trait EngineTypes:
 
 /// Type that validates an [`ExecutionPayload`].
 #[auto_impl::auto_impl(&, Arc)]
-pub trait PayloadValidator: Send + Sync + Unpin + 'static {
+pub trait PayloadValidator<State = ()>: Send + Sync + Unpin + 'static {
```

Comment:
we likely don't need this generic if we make `TreeCtx<'_, Self::Block>`

Reply:
the trait is now generic over `N: NodePrimitives`